lib/protocol/packets/RowDataPacket.js

```diff
@@ -61,7 +61,7 @@ function typeCast(field, parser, timeZone, supportBigNumbers, bigNumberStrings,
     case Types.DATETIME2:
     case Types.NEWDATE:
       var dateString = parser.parseLengthCodedString();
-      if (dateStrings) {
+      if (dateStrings === true || field.type === Types.DATE && dateStrings.toString().toLowerCase() === "date") {
```

Comment:
I'm not sure if this is backwards-compatible, because code today with `dateStrings: 1` will no longer work after this change, since we allowed any truthy value before to mean date strings.